### PR TITLE
Fix/sphinx

### PR DESCRIPTION
**Pull request by fraimondo**
* [x] description of feature/fix
* [x] tests added/passed
* [x] add an entry to the [latest changes](../docs/changes/latest.inc)

The documentation was a mess. This PR is to fix:

* references: now sphinx references are working properly and checked. Docs will not build if references are broken.
* numpydoc: followed the style.
* __init__ docstrings are gone. This created a conflict with autodoc.

### CODE REVIEW

**Comment by fraimondo** [comment]
I still don't know why my local preview looks like this (nicer than the real preview):

Screenshot 2022-11-03 at 00 50 19

**Comment by fraimondo** [comment]
Updated docs requirements and now the doc looks better!

Ready to be reviewed.

File: junifer/configs/juseless/datagrabbers/aomic_id1000_vbm.py [hunk 1]
```diff
@@ -27,7 +27,6 @@ class JuselessDataladAOMICID1000VBM(PatternDataladDataGrabber):
     """
 
     def __init__(self, datadir: Union[str, Path, None] = None) -> None:
-        """Initialize the class."""
```
Comment (synchon) [inline junifer/configs/juseless/datagrabbers/aomic_id1000_vbm.py:30]: Is flake8 not complaining about this?

File: junifer/configs/juseless/datagrabbers/camcan_vbm.py [hunk 2]
```diff
@@ -28,7 +28,6 @@ class JuselessDataladCamCANVBM(PatternDataladDataGrabber):
     """
 
     def __init__(self, datadir: Union[str, Path, None] = None) -> None:
-        """Initialize the class."""
```
Comment (synchon) [inline junifer/configs/juseless/datagrabbers/camcan_vbm.py:31]: Is flake8 not complaining about this?

File: junifer/configs/juseless/datagrabbers/ixi_vbm.py [hunk 3]
```diff
@@ -36,7 +36,6 @@ def __init__(
         datadir: Union[str, Path, None] = None,
         sites: Union[str, List[str], None] = None,
     ) -> None:
-        """Initialize the class."""
```
Comment (synchon) [inline junifer/configs/juseless/datagrabbers/ixi_vbm.py:39]: Is flake8 not complaining about this?

File: junifer/configs/juseless/datagrabbers/ukb_vbm.py [hunk 4]
```diff
@@ -28,7 +28,6 @@ class JuselessDataladUKBVBM(PatternDataladDataGrabber):
     """
 
     def __init__(self, datadir: Union[str, Path, None] = None) -> None:
-        """Initialize the class."""
```
Comment (synchon) [inline junifer/configs/juseless/datagrabbers/ukb_vbm.py:31]: Is flake8 not complaining about this?

File: junifer/data/atlases.py [hunk 5]
```diff
@@ -174,37 +172,14 @@ def load_atlas(
     path_only : bool, optional
         If True, the atlas image will not be loaded (default False).
 
-    Extra Parameters
```
Comment (synchon) [inline junifer/data/atlases.py:177]: Why are we removing this?

File: junifer/data/atlases.py [hunk 6]
```diff
@@ -174,37 +172,14 @@ def load_atlas(
...
-        magneticfield : {"3T", "7T"}, optional
-            Magnetic field (default "3T").
-    - SUIT :
-        space : {"MNI", "SUIT"}, optional
-            Space of atlas (default "MNI"). (For more information
-            see http://www.diedrichsenlab.org/imaging/suit.htm).
-
     Returns
     -------
-    niimg-like object or None
```
Comment (synchon) [inline junifer/data/atlases.py:201]: This is what `nilearn` calls it. Is it giving an issue for rendering?

File: junifer/data/atlases.py [hunk 7]
```diff
@@ -134,7 +133,6 @@ def list_atlases() -> List[str]:
     -------
     list of str
         A list with all available atlases.
-
```
Comment (synchon) [inline junifer/data/atlases.py:137]: Is numpydoc or flake8 not complaining?

File: junifer/datagrabber/aomic/id1000.py [hunk 8]
```diff
@@ -34,7 +34,6 @@ def __init__(
         datadir: Union[str, Path, None] = None,
         **kwargs,
     ) -> None:
-        """Initialize the class."""
```
Comment (synchon) [inline junifer/datagrabber/aomic/id1000.py:37]: Same comment with the flake8 thing.

File: junifer/datagrabber/aomic/piop2.py [hunk 9]
```diff
@@ -125,7 +124,7 @@ def get_elements(self):
 
         Returns
         -------
-        elements : list
```
Comment (synchon) [inline junifer/datagrabber/aomic/piop2.py:128]: For consistency, should it be `list of str`?

File: junifer/datagrabber/datalad_base.py [hunk 10]
```diff
@@ -46,17 +46,17 @@ class DataladDataGrabber(BaseDataGrabber):
...
+    BaseDataGrabber
+        For the abstract base class of any datagrabber.
+
     Notes
     -----
     By itself, this class is still abstract as the `__getitem__` method relies
     on the parent class `BaseDataGrabber.__getitem__` which is not yet
     implemented. This class is intended to be used as a superclass of a class
     with multiple inheritance.
-
```
Comment (synchon) [inline junifer/datagrabber/datalad_base.py:55]: I remember this section needed to be at last according to numpydoc, or am I mistaken?

File: junifer/datagrabber/multiple.py [hunk 11]
```diff
@@ -19,15 +19,13 @@ class MultipleDataGrabber(BaseDataGrabber):
 
     Parameters
     ----------
-    datagrabbers : list of datagrabbers
```
Comment (synchon) [inline junifer/datagrabber/multiple.py:22]: Maybe we call it `list of datagrabber-like objects`?

File: junifer/datagrabber/pattern.py [hunk 12]
```diff
@@ -34,23 +35,15 @@ class PatternDataGrabber(BaseDataGrabber):
...
-    BaseDataGrabber
-
     """
 
     def __init__(
         self,
         types: List[str],
         patterns: Dict[str, str],
         replacements: List[str],
-        **kwargs,
```
Comment (synchon) [inline junifer/datagrabber/pattern.py:51]: This is a good practice for subclassing. Any particular issue with this?

File: junifer/datagrabber/pattern_datalad.py [hunk 13]
```diff
@@ -34,9 +34,10 @@ class PatternDataladDataGrabber(DataladDataGrabber, PatternDataGrabber):
 
     See Also
     --------
-    DataladDataGrabber
-    PatternDataGrabber
-
+    DataladDataGrabber:
+        Base class for data fetching via Datalad.
```
Comment (synchon) [inline junifer/datagrabber/pattern_datalad.py:38]: Does it require a description?

File: junifer/markers/collection.py [hunk 14]
```diff
@@ -62,7 +61,7 @@ def fit(self, input: Dict[str, Dict]) -> Optional[Dict]:
 
         Parameters
         ----------
-        input
+        input : dict
```
Comment (synchon) [inline junifer/markers/collection.py:64]: For consistency, maybe we remove `input :`?

File: junifer/markers/functional_connectivity_atlas.py [hunk 15]
```diff
@@ -36,14 +36,15 @@ class FunctionalConnectivityAtlas(BaseMarker):
         :func:`junifer.stats.get_aggfunc_by_name` (default None).
     cor_method : str, optional
         The method to perform correlation using. Check valid options in
-        :func:`nilearn.connectome.ConnectivityMeasure` (default "covariance").
+        :external:class:`nilearn.connectome.ConnectivityMeasure`
```
Comment (synchon) [inline junifer/markers/functional_connectivity_atlas.py]: `:external+nilearn`?

File: junifer/markers/functional_connectivity_atlas.py [hunk 16]
```diff
@@ -36,14 +36,15 @@ class FunctionalConnectivityAtlas(BaseMarker):
...
+        (default "covariance").
     cor_method_params : dict, optional
         Parameters to pass to the correlation function. Check valid options in
-        :func:`nilearn.connectome.ConnectivityMeasure` (default None).
+        :external:class:`nilearn.connectome.ConnectivityMeasure`
```
Comment (synchon) [inline junifer/markers/functional_connectivity_atlas.py]: Ditto.

File: junifer/stats.py [hunk 17]
```diff
@@ -24,21 +23,24 @@ def get_aggfunc_by_name(
...
+        * ``winsorized_mean`` ->
+          :external+scipy:func:`scipy.stats.mstats.winsorize`
+        * ``mean`` -> :external+numpy:func:`numpy.mean`
+        * ``std`` -> :external+numpy:func:`numpy.std`
+        * ``trim_mean`` -> :external+scipy:func:`scipy.stats.trim_mean`
+
     func_params : dict, optional
         Parameters to pass to the function.
-        E.g. for 'winsorized_mean': func_params = {'limits': [0.1, 0.1]}
+        E.g. for `winsorized_mean`: ``func_params = {'limits': [0.1, 0.1]}``
```
Comment (synchon) [inline junifer/stats.py:34]: \`\`winsorized_mean\`\`"?

File: junifer/storage/pandas_base.py [hunk 18]
```diff
@@ -30,14 +30,13 @@ class PandasBaseFeatureStorage(BaseFeatureStorage):
 
     See Also
     --------
-    BaseFeatureStorage
+    BaseFeatureStorage : The base class for feature storage.
```
Comment (synchon) [inline junifer/storage/pandas_base.py:33]: The format is inconsistent with the previous instances, which one do we go for?

File: junifer/storage/sqlite.py [hunk 19]
```diff
@@ -47,7 +47,7 @@ class SQLiteFeatureStorage(PandasBaseFeatureStorage):
 
     See Also
     --------
-    PandasBaseFeatureStorage
+    PandasBaseFeatureStorage : The base class for Pandas-based feature storage.
```
Comment (synchon) [inline junifer/storage/sqlite.py:50]: Ditto.

File: tox.ini [hunk 20]
```diff
@@ -95,6 +95,7 @@ max-line-length = 79
 extend-ignore =
     B024  # abstract class with no abstract methods
     D202
+    D107  # missing docstring in __init__, incompatible with numpydoc
```
Comment (synchon) [inline tox.ini:98]: So is the issue with numpydoc or Sphinx?

**Review by synchon — CHANGES_REQUESTED** [review]

File: junifer/configs/juseless/datagrabbers/aomic_id1000_vbm.py (hunk 1 above)
Comment (fraimondo) [reply junifer/configs/juseless/datagrabbers/aomic_id1000_vbm.py:30]: D107 is incompatible with numpydoc

File: junifer/data/atlases.py (hunk 5 above)
Comment (fraimondo) [reply junifer/data/atlases.py:177]: Should not be here, there are no extra parameters in `load_atlas`

File: junifer/stats.py (hunk 17 above)
Comment (fraimondo) [reply junifer/stats.py:34]: done

File: tox.ini (hunk 20 above)
Comment (fraimondo) [reply tox.ini:98]: numpydoc asks for docstings in the class or the init, but not both

File: junifer/storage/pandas_base.py (hunk 18 above)
Comment (fraimondo) [reply junifer/storage/pandas_base.py:33]: this is numpydoc, I would go that format.

File: docs/Makefile [hunk 21]
```diff
@@ -3,7 +3,7 @@
 
 # You can set these variables from the command line, and also
 # from the environment for the first two.
-SPHINXOPTS    ?="-W"
+SPHINXOPTS    ?="-W" "-E"
```
Comment (synchon) [inline docs/Makefile:6]: Do we need this? We have `make clean`.

**Review by synchon — CHANGES_REQUESTED** [review]

**Review by synchon — APPROVED** [review]
LGTM. Thanks for the fixes! 🚀

**Pull request merged**